plugin/artifactory_client.go

```diff
@@ -185,9 +186,15 @@ func (ac *artifactoryClient) DeletePermissionTarget(ptName string) error {
 func (ac *artifactoryClient) CreateToken(tokenReq TokenCreateEntry, role *RoleStorageEntry) (auth.CreateTokenResponseData, error) {
 	expiresIn := uint(tokenReq.TTL.Seconds())
 
+	var groups []string
+	if len(role.PermissionTargets) > 0 {
+		groups = append(groups, groupName(role))
+	}
+	groups = append(groups, role.Groups...)
+
 	params := accessservices.CreateTokenParams{
 		CommonTokenParams: auth.CommonTokenParams{
-			Scope:     fmt.Sprintf("applied-permissions/groups:%s", groupName(role)),
+			Scope:     fmt.Sprintf("applied-permissions/groups:%s", strings.Join(groups, ",")),
```

Comment:
Is this guaranteed the comma is disallowed in the group name? If there's no such validation before we can get into trouble here

Reply:
There is no validation but we generate the role name and there is no way to override it from a consumer perspective.